cf-agent/nfs.c

```diff
@@ -126,14 +128,58 @@ static const char *const VMOUNTOPTS[] =
     [PLATFORM_CONTEXT_ANDROID] = "defaults",         /* android */
 };
 
+static void GetHostAndSource(const char *buf, char *host, char *source)
+/* Extracts from buffer host & source of nfs, cifs and panfs network systems */
+{
```

Comment:
I'd add `assert(StringStartsWith(buf, "panfs://") || StringStartsWith(buf, "//") || strchr(buf, ":"))` here.

Reply:
Alright. I'll leave this PR as is and do that in the master PR.

Reply:
Added in master PR: https://github.com/cfengine/core/pull/3903